docs/index.rst

```diff
@@ -17,6 +21,32 @@ More in-depth coverage is provided in other sections.
     api/index
     examples
 
+
+Concepts
+--------
+
+Besides code which implements Widgetastic and navmazing functionality, two
+basic concepts used to abstract Satellite/Foreman UI functionality are Views
+and Entities.
+
+**View** is part of page containing widgets, it describes appearance of the
```

Comment:
+**View** is part of page containing widgets, it describes appearance of the
+page. Here all the nasty XPaths goes, but please create Foreman or Katello
+pull requests for `id=...` additions where reasonable, as generally XPaths
+are too fragile.

**View** is fundamental basic concept that describes appearance of the page. Here we store all information about controls (widgets) on the page, specifically locators to them or when control is basic one that has common locator we just define widget without any further details. Of course, we try to address locators using id or name as first priority, but if we can't - we go with Xpath. If you see situation like this - create Foreman or Katello pull request to assign unique id attribute for necessary element where it is reasonable, as Xpaths are more fragile and has worse performance to interact with